bootstrap/pkg/apis/apps/configconverters/converters.go

```diff
@@ -50,61 +36,27 @@ func LoadConfigFromURI(configFile string) (*kfconfig.KfConfig, error) {
 		return nil, err
 	}
 
-	// appFile is configFile if configFile is local.
-	// Otherwise (configFile is remote), appFile points to a downloaded copy of configFile in tmp.
-	appFile := configFile
-	// If config is remote, download it to a temp dir.
+	// If config is remote, do http get.
 	if isRemoteFile {
-		// TODO(jlewi): We should check if configFile doesn't specify a protocol or the protocol
-		// is file:// then we can just read it rather than fetching with go-getter.
-		appDir, err := ioutil.TempDir("", "")
-		if err != nil {
-			return nil, fmt.Errorf("Create a temporary directory to copy the file to.")
+		var client http.Client
+		resp, err := client.Get(configFile)
+		if err != nil || resp == nil || resp.StatusCode != http.StatusOK {
+			return nil, fmt.Errorf("Failed fetching file %v: %v", configFile, err)
 		}
-		// Open config file
-		//
-		// TODO(jlewi): Should we use hashicorp go-getter.GetAny here? We use that to download
-		// the tarballs for the repos. Maybe we should use that here as well to be consistent.
-		appFile = path.Join(appDir, "tmp_app.yaml")
-
-		log.Infof("Downloading %v to %v", configFile, appFile)
-		configFileUri, err := netUrl.Parse(configFile)
-		if err != nil {
-			log.Errorf("could not parse configFile url")
-		}
-		if isValidUrl(configFile) {
-			errGet := gogetter.GetFile(appFile, configFile)
-			if errGet != nil {
-				return nil, &kfapis.KfError{
-					Code:    int(kfapis.INVALID_ARGUMENT),
-					Message: fmt.Sprintf("could not fetch specified config %s: %v", configFile, errGet),
-				}
-			}
-		} else {
-			g := new(gogetter.FileGetter)
-			g.Copy = true
-			errGet := g.GetFile(appFile, configFileUri)
-			if errGet != nil {
-				return nil, &kfapis.KfError{
-					Code:    int(kfapis.INVALID_ARGUMENT),
-					Message: fmt.Sprintf("could not fetch specified config %s: %v", configFile, err),
-				}
-			}
-		}
-	}
+		defer resp.Body.Close()
 
-	// Read contents
-	configFileBytes, err := ioutil.ReadFile(appFile)
-	if err != nil {
-		return nil, &kfapis.KfError{
-			Code:    int(kfapis.INTERNAL_ERROR),
-			Message: fmt.Sprintf("could not read from config file %s: %v", configFile, err),
-		}
+		return ioutil.ReadAll(resp.Body)
+	} else {
+		// If config is lcoal, read file.
+		return ioutil.ReadFile(configFile)
 	}
+}
 
+// KfdefByteToKfConfig takes Kfdef in bytes and convert to KfConfig
+func KfdefByteToKfConfig(KfdefByte []byte, appDir string) (*kfconfig.KfConfig, error) {
```

Comment:
Per comment above do we need to do this if we use a type interface{} ? Then we could just to type assertion?